cli/azd/pkg/containerapps/container_app.go

```diff
@@ -256,6 +285,155 @@ func (cas *containerAppService) AddRevision(
 	return nil
 }
 
+func (cas *containerAppService) waitForRevisionReady(
+	ctx context.Context,
+	subscriptionId string,
+	resourceGroupName string,
+	appName string,
+	revisionName string,
+	logProgress func(string)) error {
+	statusPrefix := "Waiting for revision to be ready"
+	logProgress(statusPrefix)
+
+	credential, err := cas.credentialProvider.CredentialForSubscription(ctx, subscriptionId)
+	if err != nil {
+		return err
+	}
+
+	// Add a policy to capture the raw response. This allows us to get fields (ex: 'properties.runningStateDetails') that are
+	// currently not available in armappcontainers.ContainerAppsRevisionsClientGetRevisionResponse
+	responseCapture := azsdk.ResponseCapturePolicy{}
+
+	revisionsClientOptions := *cas.armClientOptions
+	revisionsClientOptions.PerCallPolicies = append(revisionsClientOptions.PerCallPolicies, &responseCapture)
```

Comment:
I don't think we need a policy for getting the http response. There is a policy already in place for it:
https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore/policy#example-WithCaptureResponse

```
// policy.WithCaptureResponse provides a mechanism for obtaining an API's underlying *http.Response
	var respFromCtx *http.Response
	ctx := policy.WithCaptureResponse(context.TODO(), &respFromCtx)
	// make some client method call using the updated context
	// resp, err := client.SomeMethod(ctx, ...)
	// *respFromCtx contains the raw *http.Response returned during the client method call.
	// if the HTTP transport didn't return a response due to an error then *respFromCtx will be nil.
	_ = ctx
```

Reply:
Really really cool! Thanks for the tip and saving us more lines-of-code :)